src/test/java/com/marklogic/kafka/connect/source/AbstractIntegrationSourceTest.java

```diff
@@ -117,4 +119,24 @@ private void assertTopicAndSingleValue(List<SourceRecord> newSourceRecords, Stri
     protected Stream<ObjectNode> recordsToJsonObjects(List<SourceRecord> records) {
         return records.stream().map(record -> readJsonObject((String) record.value()));
     }
+
+    protected String injectValueIntoQuery(String userDsl, Map<String, Object> parsedConfig, String CONSTRAINT_VALUE) {
+        parsedConfig.put(MarkLogicSourceConfig.DSL_QUERY, userDsl);
+        return new DslQueryHandler(null, parsedConfig).injectConstraintIntoQuery(CONSTRAINT_VALUE);
+    }
+
+    protected void verifyQueryReturnsExpectedRows(String constraintValue, int expectedRowCount, String stringInFirstRecord, Map<String, Object> parsedConfig) {
```

Comment:
This seems to be recreating what the Task already does. IIUC, the intent is to verify how the task works with a simulated max constraint column value. If so, I think an approach that allows us to reuse the Task here is:

1. Add a protected setter on constraintValueStore in the Task class, and give it a comment of e.g. "Only for testing" (we have some methods like this already)
2. Create a TestConstraintValueStore that accepts the given constraintValue and set that on the Task after it's been started. 
3. Then call task.poll() like normal, and you should be able to make the same assertion on the records you get back.